craft_parts/executor/environment.py

```diff
@@ -157,6 +157,17 @@ def _get_global_environment(info: ProjectInfo) -> Dict[str, str]:
     if Features().enable_overlay:
         global_environment["CRAFT_OVERLAY"] = str(info.overlay_mount_dir)
 
+    if Features().enable_partitions:
+        if not info.partitions:
+            raise RuntimeError("Partitions enabled but no partitions specified.")
```

Comment:
That's a good question. I'm not sure what our standard is for errors like this. In `craft-application` I created an `ApplicationError` class that covers this use case, and here I'm using `RuntimeError` because that seems to be what we're using for errors that indicate a bug in the app.

Is this what `FeatureError` in craft-parts is for? If so, I should probably change some of the `ValueError`s to `FeatureError`s.

Reply:
this is indeed an actual crash or bad implementation on the application side

Reply:
OK, cool :-) Changing these to FeatureError then